src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.Core.cs

```diff
@@ -83,6 +85,60 @@ private partial Assembly Load(AssemblyName assemblyName, string assemblyOriginal
             return loadContext.LoadFromAssemblyName(assemblyName);
         }
 
+        /// <summary>
+        /// Is this a registered analyzer file path that the loader knows about.
+        /// 
+        /// Note: this is using real paths, not the original file paths
+        /// </summary>
+        private bool IsRegisteredAnalyzerPath(string analyzerPath)
+        {
+            CheckIfDisposed();
+
+            lock (_guard)
+            {
+                return _realToOriginalPathMap.ContainsKey(analyzerPath);
+            }
+        }
+
+        internal string? GetAssemblyLoadPath(AssemblyName assemblyName, string directory)
+        {
+            // Prefer registered dependencies in the same directory first.
+            var simpleName = assemblyName.Name!;
+            var assemblyPath = Path.Combine(directory, simpleName + ".dll");
```

Comment:
any concern that this won't find the dll since the map is keyed by Ordinal, not OrdinalIgnoreCase?  trying to figure out hte relation betweent he strign keys in memory (keyed ordinally), and the actual file paths (keyed without case on windows at least).

Reply:
There is no winning with path and case sensitivity, only picking the way you lose 😦 

The approach we generally take with the compiler, for newer code paths, is we use ordinal comparisons on inputs. Essentially we consider a.cs and a.CS to be different files. Then we let the file system do as it will with the paths. 

Note: this is for newer code paths, of which analyzers is generally one of them. If you look at the compiler you will see a variety of strategies that have no overall plan. Hard to change them though cause we know people are relying on them.

Reply:
> There is no winning with path and case sensitivity, only picking the way you lose 

haha.  totally agree.

my overall request then is that we doc somewhere in *this component area* the philosophy we're taking, and/or we doc each location as to why we believe this is the right approach.  Note: it's perfectly valid to say (in the code) that thsi might be incorrect, but we've picked a strategy and don't want to change it because or risk.

What i do *not* like is when the code doesn't say *anything* and i'm going: is this intentional?  a bug? a ticking time bomb?  etc.